analytics.go

```diff
@@ -143,7 +143,7 @@ func New(key string) *Client {
 		now:      time.Now,
 		uid:      uid,
 	}
-
+	c.startLoop()
```

Comment:
Good practices when writing Go code is to have the zero-value of a type be a valid state. In our case it's kind of hard to do because we need at least the write-key (although if not provided we could have code to pick it up from the environment or other places).

So code like this could work:
```go
client := analytics.Client{
    Key: "write-key",
}

// Now we can start using the client
```
If `startLoop` is done in the constructor we force using the constructor to create valid client objects, I believe the current implementation which lazy-loads the goroutine is more powerful.

I see the value in knowing that the goroutine was started when the constructor is called (especially for testing purposes), why not simply use `c.once.Do(c.startLoop)` here instead and not remove it from `(*Client).queue` then? That way we get the best of both approaches?